code/modules/fallout/reagents/medicines.dm

```diff
@@ -102,10 +135,14 @@ datum/reagent/medicine/bitter_drink/on_mob_life(mob/living/M)
 
 /datum/reagent/medicine/healing_powder/on_mob_life(mob/living/carbon/M)
 	if(M.getBruteLoss() == 0 && M.getFireLoss() == 0)
-		metabolization_rate = 1000 * REAGENTS_METABOLISM //instant metabolise if it won't help you, prevents prehealing before combat
-	M.adjustFireLoss(-3*REAGENTS_EFFECT_MULTIPLIER)
-	M.adjustBruteLoss(-3*REAGENTS_EFFECT_MULTIPLIER)
-	M.hallucination = max(M.hallucination, 5)
+		metabolization_rate = 1000 * REAGENTS_METABOLISM //instant metabolise if it won't help you, prevents prehealing before combat|
+	if(HAS_TRAIT(M, TRAIT_TECHNOPHOBE))
```

Comment:
This should probably be more like:
```
	var/is_technophobe = HAS_TRAIT(M, TRAIT_TECHNOPHOBE)
	var/heal_rate = (is_technophobe ? -1.5 : 3) * REAGENTS_EFFECT_MULTIPLIER
	M.adjustFireLoss(heal_rate)
	M.adjustBruteLoss(heal_rate)
	M.hallucination = max(M.hallucination, is_technophobe ? 5 : 0)
```
to avoid code duplication.

Reply:
Changed a few things up.

```
/datum/reagent/medicine/healing_powder
	var/heal_factor = -1.5 //Subtractive multiplier if you do not have the perk.
	var/heal_factor_perk = 3 //Multiplier if you have the right perk.

/datum/reagent/medicine/healing_powder/on_mob_life(mob/living/carbon/M)
	var/is_technophobe = FALSE
	var/heal_rate = (is_technophobe ? heal_factor : heal_factor_perk) * REAGENTS_EFFECT_MULTIPLIER
	if(HAS_TRAIT(M, TRAIT_TECHNOPHOBE))
		is_technophobe = TRUE
	if(M.getBruteLoss() == 0 && M.getFireLoss() == 0)
		metabolization_rate = 1000 * REAGENTS_METABOLISM //instant metabolise if it won't help you, prevents prehealing before combat|
	M.adjustFireLoss(heal_rate)
	M.adjustBruteLoss(heal_rate)
	M.hallucination = max(M.hallucination, is_technophobe ? 5 : 0)
	. = TRUE
	..()
```